diff --git a/recurring_period.info.yml b/recurring_period.info.yml
index f5378d4..eeba936 100644
--- a/recurring_period.info.yml
+++ b/recurring_period.info.yml
@@ -7,6 +7,6 @@ dependencies:
 
 # Information added by Drupal.org packaging script on 2019-06-29
 version: '8.x-1.0-alpha5'
-core: '8.x'
+core_version_requirement: ^8 || ^9
 project: 'recurring_period'
 datestamp: 1561811292
diff --git a/src/DateTime/Period.php b/src/DateTime/Period.php
index c25aa0d..f3c65d6 100644
--- a/src/DateTime/Period.php
+++ b/src/DateTime/Period.php
@@ -68,7 +68,7 @@ class Period {
    *   for the label field, it will be taken over any label from the Period
    *   object.
    *
-   * @return \Drupal\Core\Entity\Entity
+   * @return \Drupal\Core\Entity\EntityBase
    *   The unsaved entity.
    */
   public function toEntity($entity_type_id, $values = []) {
diff --git a/src/Plugin/RecurringPeriod/RecurringPeriodInterface.php b/src/Plugin/RecurringPeriod/RecurringPeriodInterface.php
index 01eba4c..9cc020d 100644
--- a/src/Plugin/RecurringPeriod/RecurringPeriodInterface.php
+++ b/src/Plugin/RecurringPeriod/RecurringPeriodInterface.php
@@ -3,7 +3,8 @@
 namespace Drupal\recurring_period\Plugin\RecurringPeriod;
 
 use Drupal\recurring_period\Datetime\Period;
-use Drupal\Component\Plugin\ConfigurablePluginInterface;
+use Drupal\Component\Plugin\ConfigurableInterface;
+use Drupal\Component\Plugin\DependentPluginInterface;
 use Drupal\Core\Plugin\PluginFormInterface;
 
 /**
@@ -38,7 +39,7 @@ use Drupal\Core\Plugin\PluginFormInterface;
  * Furthermore, a Period object can be converted to an entity: see the Period
  * class for details.
  */
-interface RecurringPeriodInterface extends ConfigurablePluginInterface, PluginFormInterface {
+interface RecurringPeriodInterface extends ConfigurableInterface, DependentPluginInterface, PluginFormInterface {
 
   /**
    * Represents an unlimited end time.
